app/controllers/logs_controller.rb

```diff
@@ -1,14 +1,46 @@
 class LogsController < ApplicationController
   def index
-    logs = UseCases::Administrator::GetAuthRequests.new(
+    if location_outside_organisation?
+      redirect_to(location_new_logs_search_path) && return
+    end
+
+    if valid_search_params.blank?
+      redirect_to(new_logs_search_path) && return
+    end
+
+    @location_name = Location.find_by(id: params[:location])&.address
```

Comment:
.find is designed to look up by id
```
Location.find(params[:location])
```

And would probably call it @location_address ?

Also wonder if we could say:
```
current_organisation.locations.find(params[:location])
```

which would negate the need to validate that the location belongs to the organisation?

Reply:
`.find` throws a RecordNotFound exception if it fails - using `find_by` instead allows us to retain the control flow.

I like your solution better though - just calling `find` and 404ing if it's not a location.

Reply:
Yeah, with a bit of a play I like this structure better. Pushed [some changes to the controller](https://github.com/alphagov/govwifi-admin/pull/404/commits/c83117a178aa650da4afb7cd71a14752f90a259a).